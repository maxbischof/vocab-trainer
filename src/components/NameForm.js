import React, { useState } from 'react'
import styled from 'styled-components'
import Button from './ui/Button'
import Input from './ui/Input'

export default function NameForm({ setName }) {
  const [nameFormInput, setNameFormInput] = useState()

  function onSubmitName(event) {
    event.preventDefault()
    setName(nameFormInput)
  }

  return (
    <Form onSubmit={onSubmitName}>
      <Input
        type="text"
        placeholder="Bitte Namen eingeben"
        onChange={(event) => setNameFormInput(event.target.value)}
      ></Input>
      <Button>Test starten</Button>
    </Form>
  )
}

const Form = styled.form`
  display: flex;
  flex-direction: column;
  align-items: center;
`
