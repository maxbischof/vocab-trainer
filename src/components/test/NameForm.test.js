import React from 'react'
import NameForm from './NameForm'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(<NameForm />).toJSON()
  expect(tree).toMatchSnapshot()
})
