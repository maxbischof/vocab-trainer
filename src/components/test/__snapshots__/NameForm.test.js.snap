// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="sc-dlfnbm lgmEEb"
  onSubmit={[Function]}
>
  <input
    className="sc-gsTCUz fPXgzG"
    onChange={[Function]}
    placeholder="Bitte Namen eingeben"
    type="text"
  />
  <button
    className="sc-bdfBwQ bNYEgh"
  >
    Test starten
  </button>
</form>
`;
